test/native/helpers.js

```diff
@@ -59,6 +59,13 @@ export async function initializeEditor( props ) {
 		},
 	} );
 
+	// In some cases, we need to wait for an element to be present before
+	// considering the initialization finished.
+	if ( waitForElement ) {
+		const element = await waitFor( () => waitForElement( renderResult ) );
+		return { ...renderResult, element };
+	}
```

Comment:
@dcalhoun I've updated the `initializeEditor` function and removed the approach of waiting for an element in [this commit](https://github.com/WordPress/gutenberg/pull/35476/commits/ef6afb2e78bfde2ff2406c15da4df3e6f45667a3), besides I posted my findings in the referenced issue https://github.com/WordPress/gutenberg/issues/35685. I'll continue testing this on other cases but I'd appreciate it if you could take a look, thanks 🙇 !

Reply:
Thanks for the ping, @fluiddot. I followed up in https://github.com/WordPress/gutenberg/issues/35685#issuecomment-961919931. 

> As far as I tested, looks like it's not recommended to use the React Native testing library in an async function as it could lead to `act` warnings, like the one described in the issue.

Also, I interpret the above quote from https://github.com/WordPress/gutenberg/issues/35685#issuecomment-958955836 to relate to https://github.com/WordPress/gutenberg/pull/35476/commits/ef6afb2e78bfde2ff2406c15da4df3e6f45667a3. Is that accurate? Was there documentation expressing the recommendation or was it something you noticed from testing? I'm not sure I follow how `async` vs `Promise` changes the outcome.

Reply:
> Also, I interpret the above quote from [#35685 (comment)](https://github.com/WordPress/gutenberg/issues/35685#issuecomment-958955836) to relate to [ef6afb2](https://github.com/WordPress/gutenberg/commit/ef6afb2e78bfde2ff2406c15da4df3e6f45667a3). Is that accurate? 

Yeah, using the `Promise` approach vs `async` is the workaround I introduced to prevent the warnings.

> Was there documentation expressing the recommendation or was it something you noticed from testing? I'm not sure I follow how `async` vs `Promise` changes the outcome.

Not really, it's just something I noticed from testing. To be honest, I haven't dug deeper to find exactly what is producing different results when using `Promise` vs `async`, actually, it could be somehow related to the custom `waitFor` function 🤔 . I'll try to reproduce this issue with the async queries from the React Native testing library, and if I manage to reproduce I'll consider opening an issue in the library.

Reply:
> To be honest, I haven't dug deeper to find exactly what is producing different results when using `Promise` vs `async`, actually, it could be somehow related to the custom `waitFor` function 🤔 .

@fluiddot I see. You may be correct about its relation to our custom `waitFor`. I also wonder if it has anything to do with how [different presets can generate different code for `async`/`await`](https://github.com/callstack/react-native-testing-library/issues/506#issuecomment-691535707). I.e. I wonder if using `Promise` avoids some issue with the Babel-generated code.

Reply:
> I also wonder if it has anything to do with how different presets can generate different code for async/await. I.e. I wonder if using Promise avoids some issue with the Babel-generated code.

It could be probably related 🤔 . Thanks for pointing out your comment about the presets, it will help me to test deeper the difference between the `Promise` and `async` usage related to the issue 🙇 .